modules/nextflow/src/main/groovy/nextflow/container/CharliecloudBuilder.groovy

```diff
@@ -78,9 +84,23 @@ class CharliecloudBuilder extends ContainerBuilder<CharliecloudBuilder> {
         return this
     }
 
+    protected String getRoot(String path) {
+        def rootPath = path.split("/")
+
+        if (rootPath.size() >= 1)
+            rootPath = "/${rootPath[1]}"
```

Comment:
Not sure about this. What is expected to take? if you want the first path component it should be `rootPath[0]`

Reply:
Yes, I just want the first path component of the path. However, those are absolute path so `rootPath[0]` is empty

```
println "/test/path".split("/")
[, test, path]
```

Reply:
use `.tokenize('/')` instead, should avoid that problem

Reply:
I made the modification in the commit  a5df098ff

Reply:
ok, wonder tho if there's a unit test for it ?

Reply:
You are right, I added two test for that in the commit 45de7b8c1

Reply:
thanks a lot